zerver/lib/hotspots.py

```diff
@@ -63,13 +63,31 @@ def to_dict(self, delay: float = 0) -> Dict[str, Union[str, float, bool]]:
 
 NON_INTRO_HOTSPOTS: List[Hotspot] = []
 
+
+@dataclass
+class OneTimeNotice:
+    name: str
+
+    def to_dict(self) -> Dict[str, str]:
+        return {
+            "name": self.name,
+        }
```

Comment:
I think we should add a `type` field to these in the API, with the possible values being `one_time_notice` and `hotspot`. That would allow you to greatly simplify the JS code for figuring out whether something is a hotspot or one time notice; the web app should be able to just look at the type field rather than hardcoding `ALL_HOTSPOTS` and determining what's a one_time_notice via `!ALL_HOTSPOTS.has(onboarding_step.name)`.

(It can be the same API feature level).

Reply:
Done